code/go/0chain.net/smartcontract/storagesc/partitions_weights.go

```diff
@@ -200,10 +200,24 @@ func (bp *blobberWeightPartitionsWrap) remove(state state.StateContextI, blobber
 
 	// reduce the weight of the replace item's partition
 	bp.partWeights.Parts[removeLocs.Replace].Weight -= int(repBw.GetWeightV2())
+
+	actErr := state.WithActivation(balances, "artemis", func() error {
+		return nil
+	}, func() error {
```

Comment:
can you please explain

Reply:
Following check will fail if all elements from last partition are removed. So if last partition doesn't have anything we should remove it from the list completely.

```
if len(bp.partWeights.Parts)-1 != removeLocs.Replace {
		return fmt.Errorf("replace item must be from the last partition")
	}
```